external-import/mandiant/src/connector/base.py

```diff
@@ -646,13 +737,21 @@ def process_message(self):
                 self.helper.connector_logger.info("Connector stop")
                 sys.exit(0)
 
+            except StateError as err:
+                self.helper.connector_logger.error(str(err))
+                if work_id is not None:
+                    self.helper.api.work.to_processed(
+                        work_id, "Failed du to connector state error", in_error=True
+                    )
```

Comment:
```python
if work_id is not None:
    self.helper.api.work.to_processed(
        work_id, "Failed du to connector state error", in_error=True
    )
    work_id = None
    break
```

- Why suggest "**work_id = None**" :
When we intercept the error (stateError) and the work_id exists, we perform a to_processed (and indeed, we see a return of this information on the front end!). However, because of the finally block, a second to_processed is executed, as at this point the work_id still exists. This is why we're considering setting the work_id to None before entering the finally block, to avoid this behavior.

- Why suggest "**break**" :
If we don't add a break when a StateError is triggered, then the same error message will be generated for each of the “active” collections. Since a problem at state level affects all collections, it would be appropriate to exit the loop (_for collection in self.mandiant_collections:_) and wait for the next connector schedule.

Reply:
Thanks for your suggestion !